src/modules/Parallax.js

```diff
@@ -75,7 +75,16 @@ class Parallax extends Component {
         }
     }
 
-    componentWillReceiveProps(nextProps) {
+    shouldComponentUpdate(nextProps, nextState) {
+        const { bgImage } = this.props;
+        const { bgImage: stateBgImage } = this.state;
+        if (nextProps.bgImage !== bgImage && nextState.bgImage === stateBgImage) {
+            return false;
+        }
+        return true;
+    }
+
+    componentDidUpdate(nextProps) {
         const { parent, bgImage, bgImageSrcSet, bgImageSizes } = nextProps;
```

Comment:
```suggestion
        const { parent, bgImage, bgImageSrcSet, bgImageSizes } = this.props;
```
@dslemay Looked into the changes and it seems as you are using the old props instead of the current ones. The first argument of `componentDidUpdate` is `prevProps`. We should rather use `this.props` in line 88, otherwise it wouldn't get updated if the parent changes.